Assets/GothicVR/Scripts/Creator/Meshes/AbstractMeshCreator.cs

```diff
@@ -487,5 +499,29 @@ protected Material GetEmptyMaterial()
 
             return material;
         }
+
+        private Material GetWaterMaterial(PxMaterialData materialData)
+        {
+            var shader = Shader.Find(waterShader);
+            Material material = new Material(shader);
+
+            // FIXME - Here we need to see if we can change materialData.waveSpeed
+
+            switch (materialData.name)
```

Comment:
You could give this a try. This is the Vector, OpenGothic is using for animation. Sqrt from Vector is definitely not the intended idea, but see for yourself: waterfallls look fine.
```suggestion
            material.SetFloat("_ScrollSpeed", -900000 * materialData.animMapDir.ToUnityVector().SqrMagnitude());
```

Reply:
That's a cool approach using proper material data. Nevertheless it still causes some problems in the swamp / water transition zone and the rivers. I would go for the hardcoded ones for now and go for a proper implementation in the near future.